homeassistant/components/scrape/config_flow.py

```diff
@@ -125,7 +127,15 @@ def validate_rest_setup(user_input: dict[str, Any]) -> dict[str, Any]:
 
 def validate_sensor_setup(user_input: dict[str, Any]) -> dict[str, Any]:
     """Validate sensor setup."""
-    return {"sensor": [{**user_input, CONF_INDEX: int(user_input[CONF_INDEX])}]}
+    return {
+        "sensor": [
+            {
+                **user_input,
+                CONF_INDEX: int(user_input[CONF_INDEX]),
+                CONF_UNIQUE_ID: uuid.uuid1(),
```

Comment:
If we add UNIQUE_ID, then I think we need to remove DeviceClass, StateClass and UnitOfMeasurement.
We should probably also make it a string:
```suggestion
                CONF_UNIQUE_ID: str(uuid.uuid1()),
```

Reply:
Why would we remove those?
Perhaps we should consider not allowing to change them (mess up statistics) but configure them initially we should do.

Reply:
If there is a unique_id, then they can be set in the frontend.
We can therefore default to None, and let the user customise afterwards (smaller config flow)

Reply:
Looks like we should remove UoM but I don't see how a user would set DeviceClass nor StateClass?

Reply:
I must be mistaken - I can't find it either anymore.
We still need to ensure we store the unique_id as a string though.